pkgs/applications/networking/instant-messengers/viber/default.nix

```diff
@@ -1,111 +1,167 @@
+            gst_all_1.gst-plugins-ugly
+            gst_all_1.gst-plugins-bad
+          ];
+        in
+        ''
+          dpkg-deb -x $src $out
+          mkdir -p $out/bin
 
-    # qt.conf is not working, so override everything using environment variables
-    wrapProgram $out/opt/viber/Viber \
-      --set QT_QPA_PLATFORM "xcb" \
-      --set QT_PLUGIN_PATH "$out/opt/viber/plugins" \
-      --set QT_XKB_CONFIG_ROOT "${xorg.xkeyboardconfig}/share/X11/xkb" \
-      --set QTCOMPOSE "${xorg.libX11.out}/share/X11/locale" \
-      --set QML2_IMPORT_PATH "$out/opt/viber/qml"
-    ln -s $out/opt/viber/Viber $out/bin/viber
+          # Soothe nix-build "suspicions"
+          chmod -R g-w $out
 
-    mv $out/usr/share $out/share
-    rm -rf $out/usr
+          # GST_PLUGIN_SYSTEM_PATH is specified because Qt6's QGStreamerMediaPlayer calls
+          # gst_element_factory_make which in turn uses the env variable to perform the lookup.
+          # Qt6 uses the returned object without checking for NULL which will lead to a SEGFAULT.
 
-    # Fix the desktop link
-    substituteInPlace $out/share/applications/viber.desktop \
-      --replace /opt/viber/Viber $out/opt/viber/Viber \
-      --replace /usr/share/ $out/share/
-  '';
+          wrapProgram $out/opt/viber/Viber \
+          --set QT_PLUGIN_PATH "$out/opt/viber/plugins" \
+          --set QML2_IMPORT_PATH "$out/opt/viber/qml" \
+          --set XDG_DATA_DIRS "out/share" \
+          --set GST_PLUGIN_SYSTEM_PATH "${gstreamerPluginPath}" \
 
-  dontStrip = true;
-  dontPatchELF = true;
+          ln -s $out/opt/viber/Viber $out/bin/Viber
 
-  meta = {
-    homepage = "https://www.viber.com";
-    description = "An instant messaging and Voice over IP (VoIP) app";
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
-    license = lib.licenses.unfree;
-    platforms = [ "x86_64-linux" ];
-    maintainers = with lib.maintainers; [ jagajaga ];
-  };
+          mv $out/usr/share $out/share
+          rm -rf $out/usr
+
+          # Fix the desktop link
+          substituteInPlace $out/share/applications/viber.desktop \
+            --replace /opt/viber/Viber Viber \
+            --replace /usr/share/ $out/share/
+        '';
+
+      preFixup = ''
+        # Ugly hack, libtiff.so.5 is not in nixpkgs. Does not seem to break anything.
+        patchelf --replace-needed libtiff.so.5 libtiff.so $out/opt/viber/plugins/imageformats/libqtiff.so
+      '';
+    };
+  package =
+    buildFHSEnv
+      {
+        name = "${pname}-${version}";
 
-}
+        inherit meta;
+        targetPkgs = pkgs: with pkgs;
+          [
+            (callPackage unwrappedPackage { })
+            # needed because of hard-coded lookup to /usr/share/X11/xkb
+            xorg.xkeyboardconfig
+            # needed because of hard-coded lookup to /usr/share/fonts
+            open-fonts
+            # needed for xdg-mime - improves startup time
+            xdg-utils
+          ];
```

Comment:
```suggestion
        targetPkgs = [
          (callPackage ./unwrapped.nix { })
          # needed because of hard-coded lookup to /usr/share/X11/xkb
          xorg.xkeyboardconfig
          # needed because of hard-coded lookup to /usr/share/fonts
          open-fonts
          # needed for xdg-mime - improves startup time
          xdg-utils
        ];
```

Reply:
Is this valid? targetPkgs seems to be a function taking pkgs as an argument. I believe this is to handle x86_64 vs x86.